src/main/java/edu/cornell/kfs/fp/document/web/struts/RecurringDisbursementVoucherAction.java

```diff
@@ -0,0 +1,110 @@
+import java.util.Map;
+import java.util.Properties;
+import java.util.Set;
+
+import javax.servlet.http.HttpServletRequest;
+import javax.servlet.http.HttpServletResponse;
+
+import org.apache.commons.lang.StringUtils;
+import org.apache.struts.action.ActionForm;
+import org.apache.struts.action.ActionForward;
+import org.apache.struts.action.ActionMapping;
+import org.kuali.kfs.sys.KFSKeyConstants;
+import org.kuali.kfs.sys.context.SpringContext;
+import org.kuali.rice.core.api.util.RiceConstants;
+import org.kuali.rice.kns.util.KNSGlobalVariables;
+import org.kuali.rice.krad.service.DocumentService;
+import org.kuali.rice.krad.util.GlobalVariables;
+import org.kuali.rice.krad.util.KRADConstants;
+import org.kuali.rice.krad.util.MessageMap;
+import org.kuali.rice.krad.util.ObjectUtils;
+import org.kuali.rice.krad.util.UrlFactory;
+
+import edu.cornell.kfs.fp.businessobject.RecurringDisbursementVoucherPDPStatus;
+import edu.cornell.kfs.fp.document.RecurringDisbursementVoucherDocument;
+import edu.cornell.kfs.fp.service.RecurringDisbursementVoucherDocumentService;
+import edu.cornell.kfs.sys.CUKFSKeyConstants;
+
+public class RecurringDisbursementVoucherAction extends CuDisbursementVoucherAction implements Serializable {
+
+    private static final long serialVersionUID = -6417266354906825200L;
+    private static final org.apache.log4j.Logger LOG = org.apache.log4j.Logger.getLogger(RecurringDisbursementVoucherAction.class);
+    private static final String dot = ".";
+    
+    private transient RecurringDisbursementVoucherDocumentService recurringDisbursementVoucherDocumentService;
+    private transient DocumentService documentService;
+    
+    @Override
+    public ActionForward route(ActionMapping mapping, ActionForm form, HttpServletRequest request, HttpServletResponse response) throws Exception {
+        RecurringDisbursementVoucherForm recurringForm = (RecurringDisbursementVoucherForm) form;
+        if (getDocumentService().getByDocumentHeaderId(recurringForm.getDocument().getDocumentNumber()) == null) {
+            GlobalVariables.getMessageMap().putErrorWithoutFullErrorPath(dot + "Save", 
+                    KFSKeyConstants.ERROR_CUSTOM, "This Document needs to be saved before Submit");
+        }
+        ActionForward forward = super.route(mapping, form, request, response);
+        return forward;
+    }
+    
+    public ActionForward confirmAndCancel(ActionMapping mapping, ActionForm form, HttpServletRequest request, HttpServletResponse response) throws Exception {
+        RecurringDisbursementVoucherForm recurringForm = (RecurringDisbursementVoucherForm) form;
+        RecurringDisbursementVoucherDocument recurringDV = (RecurringDisbursementVoucherDocument) recurringForm.getDocument();
+        
+        if (StringUtils.isEmpty(recurringDV.getPaymentCancelReason())) {
+            GlobalVariables.getMessageMap().putError("document.paymentCancelReason", CUKFSKeyConstants.ERROR_RCDV_PAYMENT_CANCEL_REASON_REQUIRED);
```

Comment:
Sorry for not noticing this before - there may be constants for "document" and "paymentCancelReason" which would be preferable to the string literal.